src/Microsoft.Azure.WebJobs.Host/Hosting/WebJobsExtensionBuilderExtensions.cs

```diff
@@ -69,8 +72,8 @@ public static class WebJobsExtensionBuilderExtensions
                 throw new ArgumentNullException(nameof(configure));
             }
 
-            builder.Services.AddSingleton<IConfigureOptions<TOptions>>(p =>
-                new WebJobsExtensionOptionsConfiguration<TOptions>(p.GetService<IConfiguration>(), builder.ExtensionInfo.ConfigurationSectionName, configure));
+            builder.Services.AddSingleton<IConfigureOptions<TOptions>>(p => new WebJobsExtensionOptionsConfiguration<TOptions>(p.GetService<IConfiguration>(), builder.ExtensionInfo, configure));
+            builder.Services.TryAddEnumerable(ServiceDescriptor.Singleton<IWebJobsExtensionOptionsConfiguration, WebJobsExtensionOptionsConfiguration<TOptions>>(p => new WebJobsExtensionOptionsConfiguration<TOptions>(p.GetService<IConfiguration>(), builder.ExtensionInfo, configure)));
```

Comment:
I wonder if a better solution which also sidesteps the instance issue is to instead register instances of a type ExtensionOptionsInfo which can be constructed using TOptions and builder.ExtensionInfo so we don't have to delay create. This way we don't have to modify the existing configure type with the new interface.

Having it be a class rather than an interface will allow us to add additional info to it if we need to in the future (interfaces make that hard).

Reply:
I introduce the change.  I can't find the interface with just add the instance. However, make it clear difference between two.